### ISSUE TITLE
AverageExpression returns infinite values

### ISSUE THREAD

**Issue by rbpatt2019** [satijalab/seurat#3106]
This was raised in #3004. The owner commented again, but the issue hasn't been re-opened, so I decided to raise a new one. 

AverageExpression(obj, assay = 'RNA', slot = 'data') returns infinite values. I can confirm that this **is not** the results of negative or NaN values. 

```r
# Some quick helper functions for use with any and all
# The first checks for negatives, the second checks for positive (greater than or equal to 0)

is.negative <- function(x){
  x < 0
}
is.positive <- function(x){
  x >= 0
}

any(is.na(obj@assays$RNA@data)) # Returns FALSE
any(is.negative(obj@assays$RNA@data)) # Returns FALSE
all(is.positive(obj@assays$RNA@data)) # Returns TRUE
````

Both `any(is.na())`and `any(is.negative())`return `FALSE`, while `all(is.positive())`returns `TRUE`, so I can fairly definitively say there are no negative or NaN values in my `obj@assays$RNA@data` slot. Thoughts?

System info:
OS: MacOS Catalina (10.15.3)
Seurat: 3.1.4
R: 3.6.2

**Comment by rbpatt2019** [comment]
The same average expression call is the one that produces infinite values. Additionally, it says some of clusters have expressions on the order of 10^250, which is patently absurd. The same genes in my manually calculated AverageExpression are highly expressed, but on a much more believable scale of 10^3. See the below screenshot, generated from the same variables as the previous comment:
![image](https://user-images.githubusercontent.com/43184541/83788377-199e4f80-a663-11ea-9d8a-0368062ec409.png)

I won't rule out the possibility I'm mis-understanding something, but, if so, I would love an explanation as to why my above to methods are not equivalent. Cheers!

**Comment by satijalab** [comment]
Thanks, we'll take a look at this (@jaisonj708) . Can you confirm that you have normalized the data with LogNormalize prior to running AverageExpression? If you have, would you be able to e-mail a copy of your Seurat object to seuratpackage@gmail.com?

**Comment by rbpatt2019** [comment]
Sorry for the late reply @satijalab and @jaisonj708 ! I got swamped last week finishing a manuscript. 

Bit more information - I used `SCTransform` before calling `AverageExpression`. That said, I can also confirm that if I create a new object, call `NormalizeData` using the LogNormalize method, I still return Infinite values and values on the order 10^250. 

After a bit more playing around, I've also discovered that the problem occurs on any Seurat object I test it on, so doesn't seem to be object specific.

I've emailed an example object to the above email, and will report back if I find anything. Thank you for the help!
